odoo/addons/base/models/res_company.py

```diff
@@ -258,6 +258,22 @@ def write(self, values):
             self.invalidate_model(company_address_fields)
         return res
 
+    @api.constrains('active')
+    def _check_active(self):
+        for company in self:
+            if not company.active:
+                sample_active_users = self.env['res.users'].search([
+                    ('company_id', '=', company.id),
+                    ('active', '=', True),
+                ], limit=10)
```

Comment:
I guess a search_count might be correct in this scenario, and giving the exact count instead of the first 10 random users in the error message too. Maybe both if the count is < 10 ?
What do you think ?

Reply:
> I guess a search_count might be correct in this scenario, and giving the exact count instead of the first 10 random users in the error message too. Maybe both if the count is < 10 ?
> What do you think ?

Well, my goal was to be as user-friendly as possible, directly giving the problematic users for user having small companies, in case they do not know how to list employees in a given company.
But then, after thinking following your comment, I realize having multi companies is already for bigger companies, so they should know how to list users part of one company.. :D.
So, I switched to the user count as you suggested.

> Something else: what happens if we activate an (inactive) user that has a inactive company ?
Just adding `active` in the existing constraint `_check_company` of `res.users` does the job, as `if user.company_id not in user.company_ids:` while be False, as `user.company_ids` will exclude archived companies.

Reply:
> Just adding `active` in the existing constraint `_check_company` of `res.users` does the job, as `if user.company_id not in user.company_ids:` while be False, as `user.company_ids` will exclude archived companies.

Nice! Don't forget to add a test for this use case, so we can't forget it in future refactoring :test_tube:

Reply:
> Nice! Don't forget to add a test for this use case, so we can't forget it in future refactoring 🧪

Added